src/selectors/layout.test.js

```diff
@@ -150,8 +151,22 @@ describe('Selectors', () => {
   });
 
   describe('getZoomPosition', () => {
-    it('returns an empty object if chartSize is unset', () => {
-      expect(getZoomPosition(mockState.lorem)).toEqual({});
+    const defaultZoom = {
+      scale: 1,
+      translateX: 0,
+      translateY: 0
+    };
+
+    it('returns default values if chartSize is unset', () => {
+      expect(getZoomPosition(mockState.lorem)).toEqual(defaultZoom);
+    });
+
+    it('returns default values if graph width/height is infinite (i.e. when no nodes are visible)', () => {
```

Comment:
`Infinity` is coming from Dagre: It's returned when the graph has 0 nodes/edges. This happens when you disable all the nodes.

I could just avoid running Dagre layout at all and return `undefined` when `nodes.length === 0`. I'd just have to add some extra checks on the functions that receive the output.

Reply:
update: have done it. See 78dbf90

Reply:
Ah well if you agree that it seems more natural, I think it would be nice to catch that `Infinity` early on if it's a small change, either like you say or swap it with `0` at source?

Reply:
Agreed. Done!